server/src/services/semanticHighlight/onSemanticTokensFull.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+/* eslint-disable @typescript-eslint/strict-boolean-expressions */
+import {
+  SemanticTokens,
+  SemanticTokensParams,
+} from "vscode-languageserver-protocol";
+import { Language, ProductionKind } from "@nomicfoundation/slang";
+import _ from "lodash";
+import { analyze } from "@nomicfoundation/solidity-analyzer";
+import semver from "semver";
+import { ServerState } from "../../types";
+import { CustomTypeHighlighter } from "./highlighters/CustomTypeHighlighter";
+import { SemanticTokensBuilder } from "./SemanticTokensBuilder";
+import { KeywordHighlighter } from "./highlighters/KeywordHighlighter";
+import { ElementaryTypeHighlighter } from "./highlighters/ElementaryTypeHighlighter";
+import { NumberHighlighter } from "./highlighters/NumberHighlighter";
+import { StringHighlighter } from "./highlighters/StringHighlighter";
+import { FunctionDefinitionHighlighter } from "./highlighters/FunctionDefinitionHighlighter";
+import { FunctionCallHighlighter } from "./highlighters/FunctionCallHighlighter";
+import { EventEmissionHighlighter } from "./highlighters/EventEmissionHighlighter";
+import { EventDefinitionHighlighter } from "./highlighters/EventDefinitionHighlighter";
+import { ContractDefinitionHighlighter } from "./highlighters/ContractDefinitionHighlighter";
+import { InterfaceDefinitionHighlighter } from "./highlighters/InterfaceDefinitionHighlighter";
+import { StructDefinitionHighlighter } from "./highlighters/StructDefinitionHighlighter";
+import { OffsetTranslator } from "./OffsetTranslator";
+import { visit } from "./slangHelpers";
+
+let lastValidResponse: SemanticTokens = { data: [] };
+
+export function onSemanticTokensFull(serverState: ServerState) {
+  return (params: SemanticTokensParams): SemanticTokens => {
```

Comment:
Should we wrap this in `onCommand` or a sentry transaction?

Reply:
Good idea, will do it now